src/components/screens/vacation-requests/vacation-requests-screen.tsx

```diff
@@ -0,0 +1,287 @@
+import { Card } from "@mui/material";
+import { useEffect, useState } from "react";
+import VacationRequestsTable from "../../vacation-requests-table/vacation-requests-table";
+import {
+  VacationRequest,
+  VacationRequestStatus,
+  VacationRequestStatuses
+} from "../../../generated/client";
+import { useApi } from "../../../hooks/use-api";
+import { useAtom, useAtomValue, useSetAtom } from "jotai";
+import { userProfileAtom } from "../../../atoms/auth";
+import { errorAtom } from "../../../atoms/error";
+import { GridRowId } from "@mui/x-data-grid";
+import { VacationData } from "../../../types";
+import { vacationRequestsAtom } from "../../../atoms/vacationRequests";
+import { vacationRequestStatusesAtom } from "../../../atoms/vacationRequestStatuses";
+import strings from "../../../localization/strings";
+
+/**
+ * Vacation requests screen
+ */
+const VacationRequestsScreen = () => {
+  const { vacationRequestsApi, vacationRequestStatusApi } = useApi();
+  const userProfile = useAtomValue(userProfileAtom);
+  const setError = useSetAtom(errorAtom);
+  const [vacationRequests, setVacationRequests] = useAtom(vacationRequestsAtom);
+  const [vacationRequestStatuses, setVacationRequestStatuses] = useState<VacationRequestStatus[]>(
+    []
+  );
+  const setLatestVacationRequestStatuses = useSetAtom(vacationRequestStatusesAtom);
+  const [isLoading, setIsLoading] = useState(true);
```

Comment:
loading not isLoading as mentioned in previous comments

Reply:
Loading states should be initialized as false. Within each function that is making an async request you should set the loading state to true within the try and set the loading state back to false after the catch. Happy to discuss if needed.

Reply:
I got rid of the loading state. Having a separate loading state for the DataGrid was too tricky especially when having the loading state changing on every async request causing the custom loading overlay to flicker on initial loading of DataGrid rows. Instead I opted for this and it works with the custom loading overlay how I originally intended:
`
<DataGrid>
  ...
  loading={rows.length ? false : true}
  ...
</DataGrid>
`

Reply:
Initialized loading state as false and added setLoading to each async request. Had to change useEffect to useMemo as a result for fetching requests and statuses and had to change loading state logic for the DataGrid: `loading={loading && !rows.length ? true : false}`.